### PR TITLE
add min insertions & deletions example

### PR DESCRIPTION
**Pull request by ankit-bagde**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: dynamic_programming/min_insertions_deletions.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+Example, given str1 = "foam" and str2 = "programming"
+
+LCS(str1, str2) = L = "oam"  => length(L) = 3
+Thus,
+deletions = len(str1) - len(L), 
+insertions = len(str2) - len(L)
+"""
+
+
+def longest_common_subsequence(str1: str, str2: str):
```
Comment (xcodz-dot) [inline dynamic_programming/min_insertions_deletions.py:18]:
```suggestion
def longest_common_subsequence(str1: str, str2: str) -> int:
```

add type hinting for return value

File: dynamic_programming/min_insertions_deletions.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    (6, 2)
+    >>> min_insertions_deletions("applaud", "app")
+    (0, 4)
+    >>> min_insertions_deletions(32, "app")
+    Traceback (most recent call last):
+      File "min_insertions_deletions.py", line 96, in <module>
+        insertions, deletions = min_insertions_deletions(str1, str2)
+      File "min_insertions_deletions.py", line 78, in min_insertions_deletions
+        assert type(str1) is str, "str1 should be a string, not {}".format(type(str1))
+    AssertionError: str1 should be a string, not <class 'int'>
```
Comment (xcodz-dot) [inline dynamic_programming/min_insertions_deletions.py:84]:
```suggestion
    >>> min_insertions_deletions(32, "app")
    Traceback (most recent call last):
      ...
    AssertionError: str1 should be a string, not <class 'int'>
```

details are not necessary in doctests

**Review by xcodz-dot — CHANGES_REQUESTED** [review]
These are few changes (enhancements). Also please fix your error of trailing whitespaces using a code formatter like `black`. You can read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md) for instructions to use black.

**Comment by Ankit-2000** [comment]
Thank you for the suggestions. I have committed the changes

**Review by xcodz-dot — APPROVED** [review]
@dhruvmanila @poyea @Kush1101, check this out (ready to be merged)

**Pull request closed without merging**